packages/arb-validator/rollup/opinionThread.go

```diff
@@ -197,8 +197,14 @@ func (chain *ChainObserver) startOpinionUpdateThread(ctx context.Context) {
 					if isPrepared && chain.nodeGraph.leaves.IsLeaf(chain.calculatedValidNode) {
 						lowerBoundBlock := prepared.params.TimeBounds.LowerBoundBlock
 						upperBoundBlock := prepared.params.TimeBounds.UpperBoundBlock
+						lowerBoundTime := prepared.params.TimeBounds.LowerBoundTimestamp
+						upperBoundTime := prepared.params.TimeBounds.UpperBoundTimestamp
 						endCushion := common.NewTimeBlocks(new(big.Int).Add(chain.latestBlockId.Height.AsInt(), big.NewInt(3)))
-						if chain.latestBlockId.Height.Cmp(lowerBoundBlock) >= 0 && endCushion.Cmp(upperBoundBlock) <= 0 {
+
+						if chain.latestBlockId.Height.Cmp(lowerBoundBlock) >= 0 &&
+							endCushion.Cmp(upperBoundBlock) <= 0 &&
+							time.Now().Unix() >= lowerBoundTime.Int64() &&
```

Comment:
This was mostly laziness on my part. I think this should be safe, but I did it because the block timestamp wasn't available in this part of the code

Reply:
Discussed offline, we're predicting what the timestamp will be when we submit which is likely to be close to the current time rather than the time of the previous block

Reply:
Added comment to this effect